messaging/src/main/java/org/axonframework/queryhandling/FluxSinkWrapper.java

```diff
@@ -18,59 +18,53 @@
 
 import reactor.core.Disposable;
 import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Sinks;
 
 /**
- * Wrapper around {@link FluxSink}. Since project-reactor is not a required dependency in this Axon version, we need
+ * Wrapper around {@link Sinks.Many}. Since project-reactor is not a required dependency in this Axon version, we need
  * wrappers for backwards compatibility. As soon as dependency is no longer optional, this wrapper should be removed.
  *
  * @param <T> The value type
  * @author Milan Savic
+ * @author Stefan Dragisic
  * @since 3.3
  */
 class FluxSinkWrapper<T> {
 
-    private final FluxSink<T> fluxSink;
+    private final Sinks.Many<T> fluxSink;
 
     /**
      * Initializes this wrapper with delegate sink.
      *
      * @param fluxSink Delegate sink
      */
-    FluxSinkWrapper(FluxSink<T> fluxSink) {
+    FluxSinkWrapper(Sinks.Many<T> fluxSink) {
         this.fluxSink = fluxSink;
     }
 
     /**
      * Wrapper around {@link FluxSink#complete()}.
      */
     public void complete() {
-        fluxSink.complete();
+        fluxSink.tryEmitComplete();
     }
 
     /**
-     * Wrapper around {@link FluxSink#next(Object)}.
+     * Wrapper around {@link Sinks.Many#tryEmitNext(Object)}
      *
      * @param value to be passed to the delegate sink
      */
     public void next(T value) {
-        fluxSink.next(value);
+        fluxSink.tryEmitNext(value).orThrow();
```

Comment:
Did the documentation of Project Reactor state that `FluxSink#next` should be replaced by `Sinks.Many#tryEmitNext()#orThrow()`? Specifically, the `orThrow()` invocation should be validated, as we'd otherwise be changing behaviour.

Reply:
Sound judgement here @stefanvozd, thanks for elaborating :+1: